packages/wrapped-keys/src/lib/api/batch-generate-private-keys.ts

```diff
@@ -41,34 +41,40 @@ export async function batchGeneratePrivateKeys(
     pkpSessionSigs,
   });
 
-  const results = await Promise.all(
-    actionResults.map(
-      async (result): Promise<BatchGeneratePrivateKeysActionResult> => {
-        const { generateEncryptedPrivateKey, network } = result;
+  const keyParamsBatch = actionResults.map((keyData) => {
+    const { generateEncryptedPrivateKey, network } = keyData;
+    return {
+      ...generateEncryptedPrivateKey,
+      keyType: getKeyTypeFromNetwork(network),
+    };
+  });
+
+  const { ids } = await storePrivateKeyBatch({
+    pkpAddress,
+    sessionSig,
+    storedKeyMetadataBatch: keyParamsBatch,
+    litNetwork: litNodeClient.config.litNetwork,
+  });
 
-        const signature = result.signMessage?.signature;
+  const results = actionResults.map(
+    (actionResult, ndx): BatchGeneratePrivateKeysActionResult => {
+      const {
+        generateEncryptedPrivateKey: { memo, publicKey },
+      } = actionResult;
+      const id = ids[ndx]; // Result of writes is in same order as provided
```

Comment:
What do the ids mean and can you remind me what is it used for? In the same order means EVM first then Solana, how is it determined or does it even matter a long as the process is atomic?

Reply:
The id is a UUIDv4 that we generate when storing into DynamoDB -- it has no direct meaning, it's just used as a unique identifier for that persisted key from then on.  When a consumer uses the `list` method on the wrapped keys client, it returns an array of all keys persisted for the session sig's PKP, including the ID.  When someone needs the encrypted data (dataToEncryptHash, cipherText) for a particular key, they the the ID of the key they want to the `getEncryptedKey()` method :+1: